src/components/[guild]/messages/SendNewMessage/SendNewMessage.tsx

```diff
@@ -53,128 +97,154 @@ const SendNewMessage = (props: ButtonProps) => {
     formState: { errors },
     reset,
     handleSubmit,
+    setValue,
   } = methods
-
-  const { isOpen, onOpen, onClose } = useDisclosure()
-  const { isStuck } = useIsTabsStuck()
-  const { textColor, buttonColorScheme } = useThemeContext()
-
   const { onSubmit, isLoading } = useSendMessage(() => {
     reset()
     onClose()
   })
 
   const roleIds = useWatch({ control, name: "roleIds" })
+  const protocol = useWatch({ control, name: "protocol" })
+
   const { targetedCount, isTargetedCountValidating } = useTargetedCount(roleIds)
   const { data: reachableUsers, isValidating: isReachableUsersLoading } =
-    useReachableUsers("WEB3INBOX", "ROLES", roleIds)
+    useReachableUsers(protocol, "ROLES", roleIds)
+
+  const { keys: xmtpKeys } = useGetXmtpKeys()
+  const saveXmtpKeys = useSaveXmtpKeys()
+  const { subscribeXmtp } = useSubscribeXmtp()
 
   const greenTextColor = useColorModeValue("green.600", "green.300")
 
   return (
     <>
-      <Button
-        leftIcon={<Chat />}
-        onClick={onOpen}
-        {...(!isStuck && {
-          color: textColor,
-          colorScheme: buttonColorScheme,
-        })}
-        {...props}
-      >
-        New message
-      </Button>
+      <ModalOverlay />
```

Comment:
This pattern is pretty strange for me, I think we can leave the button, modal & modal content in the same component and just import the component dynamically only for admins

Reply:
I think this page is just loaded when the user is administrator. And all implementation is dynamic.